TESTS/API/SPI/SPI.cpp

```diff
@@ -58,28 +59,59 @@ void test_object()
 // Test for SD Card being present on the shield
 void test_card_present()
 {
-    SDFileSystem sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS,"sd");
-    FILE *File = fopen("/sd/card-present.txt", "w");   // open file
+    int error = 0; 
+	SDBlockDevice sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS);
+	FATFileSystem fs("sd");
+	sd.init();
+
+	error = fs.mount(&sd);
+	TEST_ASSERT_MESSAGE(error==0,"SD file system mount failed.");
+	
+    FILE *File = fopen("/sd/card-present.txt", "w+");   // open file
+	
     TEST_ASSERT_MESSAGE(File != NULL,"SD Card is not present. Please insert an SD Card.");
+
+	error = fs.unmount();
+	TEST_ASSERT_MESSAGE(error==0,"SD file system unmount failed.");
+
+	sd.deinit();
 }
 
 // Test SD Card Write
 void test_sd_w()
 {
-    SDFileSystem sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS,"sd");
+	int error = 0; 
+	SDBlockDevice sd(MBED_CONF_APP_SPI_MOSI, MBED_CONF_APP_SPI_MISO, MBED_CONF_APP_SPI_CLK, MBED_CONF_APP_SPI_CS);//alvin add
+	FATFileSystem fs("sd", &sd);
+
+	sd.init();
+
+	error = fs.mount(&sd);
+
     FILE *File = fopen("/sd/test_sd_w.txt", "w");   // open file
     TEST_ASSERT_MESSAGE(File != NULL,"SD Card is not present. Please insert an SD Card.");
     TEST_ASSERT_MESSAGE(fprintf(File, SD_TEST_STRING) > 0,"Writing file to sd card failed");             // write data
     // TODO: test that the file was written correctly
     fclose(File);                              // close file on SD
     TEST_ASSERT(true);
+
+	error = fs.unmount();

```

Comment:
can you please fix the formatting (using tabs on some lines?) line 98, line 107, line 87. Please check entire patch